src/repositories/video/video.repository.ts

```diff
@@ -76,44 +105,117 @@ export class VideoRepository {
     tags,
     community,
     language,
-    videoUploadLink
+    videoUploadLink,
+    beneficiaries,
+    permlink,
   }: {
-    video_id: string;
-    user: { 
-      sub: string; 
-      username: string;
-    };
-    title: string;
-    description: string;
-    tags: string[];
-    community: string;
-    language: string;
-    videoUploadLink: string;
+    video_id: string
+    user: {
+      sub: string
+      username: string
+      id?: string
+    }
+    title: string
+    description: string
+    tags: string[]
+    community: string
+    language: string
+    videoUploadLink: string
+    beneficiaries: string
+    permlink: string
   }): Promise<Video> {
     return await this.videoModel.create({
       video_id,
       owner: user.username,
       title: title,
       description,
-      beneficiaries: [],
+      beneficiaries: beneficiaries,
+      permlink: permlink,
+      originalFilename: '',
+      filename: '',
+      size: 0,
       tags: tags || [],
       community,
       language: language || 'en',
       video_details: {
         duration: 0,
       },
       posting_options: {
-        publish_type: "immediate",
-        publish_date: null
+        publish_type: 'immediate',
+        publish_date: null,
       },
-      created_by: user.sub,
+      created_by: user.id || user.sub,
       expires: moment().add('1', 'day').toDate(),
       upload_links: {
-        video: videoUploadLink
+        video: videoUploadLink,
       },
-      network: "hive",
+      network: 'hive',
       __flags: [],
-      __v: '0.1'
-    });
+      __v: '0.1',
+    })
+  }
+
+  async updateHiveVideoPost({
```

Comment:
is this the only time the Video model is updated? it may help to describe what stage its being updated at or whats being updated